cg/services/deliver_files/deliver_files_service/deliver_files_service_factory.py

```diff
@@ -0,0 +1,140 @@
+    DeliverFilesService,
+)
+from cg.services.deliver_files.deliver_files_service.exc import DeliveryTypeNotSupported
+from cg.services.deliver_files.delivery_file_tag_fetcher_service.delivery_file_tag_fetcher_service import (
+    FetchDeliveryFileTagsService,
+)
+from cg.services.deliver_files.delivery_file_tag_fetcher_service.sample_and_case_delivery_tags_fetcher import (
+    SampleAndCaseDeliveryTagsFetcher,
+)
+from cg.services.deliver_files.delivery_file_fetcher_service.analysis_delivery_file_fetcher import (
+    AnalysisDeliveryFileFetcher,
+)
+from cg.services.deliver_files.delivery_file_fetcher_service.delivery_file_fetcher_service import (
+    FetchDeliveryFilesService,
+)
+from cg.services.deliver_files.delivery_file_fetcher_service.fastq_and_analysis_delivery_file_fetcher import (
+    FastqAndAnalysisDeliveryFileFetcher,
+)
+from cg.services.deliver_files.delivery_file_fetcher_service.fastq_delivery_file_fetcher import (
+    FastqDeliveryFileFetcher,
+)
+from cg.services.deliver_files.delivery_file_formatter_service.delivery_file_formatter import (
+    DeliveryFileFormatter,
+)
+
+from cg.services.deliver_files.delivery_file_formatter_service.delivery_file_formatting_service import (
+    DeliveryFileFormattingService,
+)
+from cg.services.deliver_files.delivery_file_formatter_service.utils.case_file_formatter import (
+    CaseFileFormatter,
+)
+from cg.services.deliver_files.delivery_file_formatter_service.utils.sample_file_concatenation_formatter import (
+    SampleFileConcatenationFormatter,
+)
+from cg.services.deliver_files.delivery_file_formatter_service.utils.sample_file_formatter import (
+    SampleFileFormatter,
+)
+from cg.services.deliver_files.delivery_file_mover_service.delivery_file_mover import (
+    DeliveryFilesMover,
+)
+from cg.services.deliver_files.delivery_rsync_service.delivery_rsync_service import (
+    DeliveryRsyncService,
+)
+from cg.store.store import Store
+
+
+class DeliveryServiceFactory:
+    """Class to build the delivery services based on workflow and delivery type."""
+
+    def __init__(
+        self,
+        store: Store,
+        hk_api: HousekeeperAPI,
+        rsync_service: DeliveryRsyncService,
+        tb_service: TrailblazerAPI,
+        analysis_service: AnalysisService,
+    ):
+        self.store = store
+        self.hk_api = hk_api
+        self.rsync_service = rsync_service
+        self.tb_service = tb_service
+        self.analysis_service = analysis_service
+
+    @staticmethod
+    def _get_file_tag_fetcher(delivery_type: DataDelivery) -> FetchDeliveryFileTagsService:
+        """Get the file tag fetcher based on the delivery type."""
+        service_map: dict[DataDelivery, FetchDeliveryFileTagsService] = {
+            DataDelivery.FASTQ: SampleAndCaseDeliveryTagsFetcher,
+            DataDelivery.ANALYSIS_FILES: SampleAndCaseDeliveryTagsFetcher,
+            DataDelivery.FASTQ_ANALYSIS: SampleAndCaseDeliveryTagsFetcher,
```

Comment:
What is the point of this distinction if all delivery types have the same tag fetcher?

Reply:
expandability for the future, just need to add one here in case we need it

Reply:
But the tags depend on the workflow, not on data delivery. I don't see this being expanded in that direction (soon).

Reply:
resolved in person